### ISSUE TITLE
_R_CHECK_LENGTH_1_LOGIC2_ error in findBreakpointSteps

### ISSUE THREAD

**Issue by gadenbuie** [rstudio/rstudio#5268]
FYI that these lines in `.rs.findBreakpointSteps` throw an error when `_R_CHECK_LENGTH_1_LOGIC2_=TRUE`.

https://github.com/rstudio/rstudio/blob/8af730409bb6d651cc8f6816d136bea91441e7a4/src/cpp/session/modules/SessionBreakpoints.R#L120-L121

(I'm guessing this is where it happens because it's the only `&&` in `.rs.findBreakpointSteps`.)

To reprex, running the [miniUI tabs demo](https://github.com/rstudio/miniUI/blob/master/examples/tabs.R) gives the following error and traceback.

```
Listening on http://127.0.0.1:5654
Warning: Error in &&: 'length(x) = 3 > 1' in coercion to 'logical(1)'
  64: .rs.findBreakpointSteps
  63: .rs.findBreakpointSteps
  62: .rs.findBreakpointSteps
  61: .rs.findBreakpointSteps
  60: .rs.findBreakpointSteps
  59: .rs.findBreakpointSteps
  58: .rs.findBreakpointSteps
  57: .rs.removeBreakpoints
  56: registerShinyDebugHook
  55: registerDebugHook
  54: wrapFunctionLabel
  53: .subset2(public_bind_env, "initialize")
  52: Observer$new
  51: observe
  50: observeEvent
  49: serverFunc
  48: server
Error in is.call(funBody[[idx]]) && as.character(funBody[[idx]][[1]]) ==  : 
  'length(x) = 3 > 1' in coercion to 'logical(1)'
```

**Comment by ronblum** [comment]
@gadenbuie Thank you for raising the issue! We'll review it as we continue development of RStudio.

Side note: I really like the `tidyexplain` visual representations.

**Comment by gadenbuie** [comment]
Thanks @ronblum!

**Comment by ncalies** [comment]
Customer issue - #39940

**Comment by dfalty** [comment]
Verified fixed 1.3.867

**Comment by SebiJos** [comment]
Hi guys! I'm new to RStudio, thing were running quite well with Shiny... and since today, nothing works anymore:
Error in is.call(funBody[[idx]]) && as.character(funBody[[idx]][[1]]) ==  : 
  'length = 3' in coercion to 'logical(1)'
In addition: Warning message:
R graphics engine version 16 is not supported by this version of RStudio. The Plots tab will be disabled until a newer version of RStudio is installed. 
I tried many things, used basic examples, nothing helps, but I found this somehow older thread: https://github.com/rstudio/shiny/issues/2656

How do I install such a patch? could not find by myself though long time searching.

Any help would be greatly appreciated :)
